### PR TITLE
Convert Java files to Kotlin

### PR DESCRIPTION
**Pull request by MGaetan89**
Here is the `demo-cat-gallery` app converted to Kotlin.

I tried to keep each files history, but as soon as I pushed my changes, the "renames" became "add/delete" combinations. Do you have an idea you to fix that?
I tired using multiple commits, as follow, but that did not help:
- 1st commit: move Java files to the Kotlin source set
- 2nd commit: change Java files extension
- 3rd commit: switch implementation to Kotlin

### CODE REVIEW

File: demo-cat-gallery/build.gradle [hunk 1]
```diff
@@ -36,11 +37,15 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+    sourceSets {
+        main.java.srcDirs = ['src/main/kotlin']
```
Comment (thagikura) [inline demo-cat-gallery/build.gradle]: There will be only kotlin files under the srcDires. So let's remove this and use the default src/main/java srcDirs and move the kotlin files under  src/main/java

File: demo-cat-gallery/src/main/kotlin/com/google/android/flexbox/apps/catgallery/CatViewHolder.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import android.view.View
+import android.widget.ImageView
+import com.google.android.flexbox.FlexboxLayoutManager
+
+/**
+ * ViewHolder that represents a cat image.
+ */
+internal class CatViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+
+    private val mImageView = itemView.findViewById(R.id.imageview) as ImageView
```
Comment (thagikura) [inline demo-cat-gallery/src/main/kotlin/com/google/android/flexbox/apps/catgallery/CatViewHolder.kt]: Please rename this to imageView to follow the Kotlin coding conventions https://github.com/JetBrains/kotlin-web-site/blob/64218469db3f746ab1f4a1c4c569c21d654d665f/pages/docs/reference/coding-conventions.md

File: demo-cat-gallery/src/main/kotlin/com/google/android/flexbox/apps/catgallery/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val toolbar = findViewById(R.id.toolbar) as Toolbar
+        setSupportActionBar(toolbar)
+
+        val layoutManager = FlexboxLayoutManager(this)
+        layoutManager.flexWrap = FlexWrap.WRAP
```
Comment (thagikura) [inline demo-cat-gallery/src/main/kotlin/com/google/android/flexbox/apps/catgallery/MainActivity.kt]: Let's use apply to shorten the initialization

```kotlin
     val flexboxLayoutManager = FlexboxLayoutManager(this).apply {
            flexWrap = FlexWrap.WRAP
            flexDirection = FlexDirection.ROW
            alignItems = AlignItems.STRETCH
        }
```

File: demo-cat-gallery/src/main/kotlin/com/google/android/flexbox/apps/catgallery/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        layoutManager.flexDirection = FlexDirection.ROW
+        layoutManager.alignItems = AlignItems.STRETCH
+
+        val recyclerView = findViewById(R.id.recyclerview) as RecyclerView
```
Comment (thagikura) [inline demo-cat-gallery/src/main/kotlin/com/google/android/flexbox/apps/catgallery/MainActivity.kt]: Let's use apply to shorten the initialization

```kotlin
val recyclerView = findViewById(R.id.recyclerview) as RecyclerView
        recyclerView.apply {
            layoutManager = flexboxLayoutManager
            adapter = CatAdapter()
        }
```

**Comment by MGaetan89** [comment]
I pushed some changes according to your comments.

**Comment by thagikura** [comment]
LGTM. Thanks!

**Pull request merged**